### PR TITLE
KS-10/Add sidebar functionality

### PR DESCRIPTION
**Pull request by shrijan00003**
- router added
- transaction groud is added but halts for now (we can use or remove)
- based on #8 
- sidebar data item (can be changed to JSON if we will not use icon in the child one)
- sidebar, menu-item, and sub-item are added

### CODE REVIEW

File: src/app/app.route.tsx [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+import { Switch, Route } from 'react-router-dom';
+import { DashboardView } from 'dashboard/views';
```
Comment (akasrai) [inline src/app/app.route.tsx]: Ladder?

File: src/layouts/menu-dashboard.layout.tsx [hunk 2]
```diff
@@ -1,10 +1,13 @@
 import React from 'react';
+import Sidebar from 'sidebar/views/sidebar';
 interface Props {}
```
Comment (akasrai) [inline src/layouts/menu-dashboard.layout.tsx:3]: to type file, these type of common interface can be stored in app.type

File: src/sidebar/sidebar.data.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    title: 'ORDER',
+    route: '/order',
+    component: 'Order',
+  },
+  promotion: {
+    title: 'PROMOTION',
+    route: '/promotion',
+    component: 'Promotion',
+  },
+};
```
Comment (akasrai) [inline src/sidebar/sidebar.data.ts:46]: These are actually `routes-path` of app. Why not add them as `app.routes-path.ts`?

File: src/sidebar/sidebar.types.ts [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+export interface DashboardMenuItemProps {
+  props: {
+    icon: React.ReactNode;
+    name: string;
+    key: MenuKey;
+    toggleExpansion: (key: MenuKey) => void;
+    chevronIcon: (key: MenuKey) => void;
+    isExpanded: (key: MenuKey) => boolean;
+    child: React.ReactNode;
+  };
```
Comment (akasrai) [inline src/sidebar/sidebar.types.ts]: I used to align these kinda data in ladder format too unless they are dependent.

File: src/sidebar/views/menu-item.tsx [hunk 5]
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import { DashboardMenuItemProps } from 'sidebar/sidebar.types';
+import { useTranslation } from 'react-i18next';
+function MenuItem({ props }: DashboardMenuItemProps) {
+  const { t } = useTranslation();
+  const ct = (text: string) => t(`common.${text}`);
+  const isExpanded = props.isExpanded(props.key);
+  return (
```
Comment (akasrai) [inline src/sidebar/views/menu-item.tsx:11]: Adding a new line before every `return` statement makes code cleaner and will be easy to find out whats being returned at what place.

File: src/sidebar/views/menu-sub-item.tsx [hunk 6]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+import { useTranslation } from 'react-i18next';
+import { SidebarItemType } from 'sidebar/sidebar.types';
+
+const i = (Icon: any) => (Icon ? <Icon /> : null);
+function MenuSubItem({ item }: SidebarItemType) {
```
Comment (akasrai) [inline src/sidebar/views/menu-sub-item.tsx:7]: Also a new line before new function definition.

File: src/sidebar/views/menu-sub-item.tsx [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+  const { t } = useTranslation();
+  const ct = (text: string) => t(`common.${text}`);
+  return (
```
Comment (akasrai) [inline src/sidebar/views/menu-sub-item.tsx:12]: new line before return?

File: src/sidebar/views/sidebar.tsx [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+    key: MenuKey.DASHBOARD,
+    child: <SidebarItem item={dashboard} />,
+  },
+  food: {
+    name: 'FOOD',
+    key: MenuKey.FOOD,
+    icon: <AiFillLayout />,
+    child: food.map((item: SidebarPropertyType, index: number) => <SidebarItem item={item} key={index} />),
+  },
+};
```
Comment (akasrai) [inline src/sidebar/views/sidebar.tsx]: These data eventually grows to larger number increasing the number lines in this file and make code dirty. Why not move them to `sidebar.menu-items.ts` where we add and remove these kinda data.

**Review by akasrai — CHANGES_REQUESTED** [review]
I believe in clean coding and I have commented some clean coding rules that I follow. If you do like them we should follow them to make our code look impressive. 
>Write code that is easy to delete 

I like this quote very much. 😆😁

**Comment by shrijan00003** [comment]
@akasrai  
I am writing this, for addressing all of your valuable comments.
I also love ladder rule and all of them will be addressed before creating PR from DPR

File: src/sidebar/sidebar.data.ts (hunk 3 above)
Comment (shrijan00003) [reply src/sidebar/sidebar.data.ts:46]: that also sounds ok to me, my initial thought was to maintain route fro sidebar only, I will move them

**Review by akasrai — APPROVED** [review]

**Pull request merged**